src/arrayMethodSort.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compare =
+      compareFunction ||
+      function compareFunction2(x, y) {
+        return String(x) > String(y);
+      };
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
```

Comment:
The bubble sort algorithm is correct but not efficient for large arrays. If performance is a concern, consider implementing a more efficient sorting algorithm like quicksort or mergesort.